console/v1/types_console_sample.go

```diff
@@ -0,0 +1,166 @@
+package v1
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// ConsoleSample is an extension to customizing OpenShift web console by adding samples.
+//
+// Compatibility level 2: Stable within a major release for a minimum of 9 months or 3 minor releases (whichever is longer).
+// +openshift:compatibility-gen:level=2
+type ConsoleSample struct {
+	metav1.TypeMeta `json:",inline"`
+
+	// metadata is the standard object's metadata.
+	// More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#metadata
+	metav1.ObjectMeta `json:"metadata"`
+
+	// +kubebuilder:validation:Required
+	// +required
+	Spec ConsoleSampleSpec `json:"spec"`
+}
+
+// ConsoleSampleSpec is the desired sample for the web console.
+// Samples will appear with their title, descriptions and a badge in a samples catalog.
+type ConsoleSampleSpec struct {
+	// title is the display name of the sample.
+	//
+	// It is required and must be no more than 50 characters in length.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=50
+	// +required
+	Title string `json:"title"`
+
+	// abstract is a one-liner that describes the sample.
```

Comment:
```suggestion
	// abstract is a one line description that describes the sample.
```
Should also explain the max length in prose. Is 100 chars enough?